js/bruh-dash.js

```diff
@@ -127,14 +285,49 @@ global.bruhdash = {
 
   // iterates over elements of a collection and returns all elements that the predicate returns truthy for
   // Note: this should work for arrays and objects
-  filter: function() {
-
+  //returns an array of filtered elements - all the elements that passed the verification condition/function
+
+  filter: function(col, func) {
+    var newEl;
+    var newArr = [];
+    if (Array.isArray(col)){
+      for (var i = 0; i < col.length; i++){
+        if (func(col[i])){
+          newEl = col[i];
+          newArr.push(newEl);
+        }
+      }
+    } else if(typeof col === 'object'){
+        for (var key in col){
+          if (func(col[key]) === true){
+            newEl = col[key];
+            newArr.push(newEl);
+          }
+        }
+    }
+    return newArr;
+    //is this an array of objects or properties of a single object?
+    //what is it returning? - an array? 
+    //the intention of this function when it comes to objects isn't clear to me
```

Comment:
It is always returning an array, the items in the array should be the key/value pairing of the object where the value is truthy.